samples: posix: gettimeofday does not build on native_posix

**Describe the bug**
The samples/posix/gettimeofday sample has this in the readme:
```
For comparison, a version for native POSIX operating systems (e.g. Linux)
can be built using:

.. code-block:: console

   make -f Makefile.posix
```

However, this does not work:
```
cc src/main.c -o gettimeofday
src/main.c: In function ‘main’:
src/main.c:20:30: warning: implicit declaration of function ‘time’ [-Wimplicit-function-declaration]
   20 |                 time_t now = time(NULL);
      |                              ^~~~
src/main.c:13:1: note: ‘time’ is defined in header ‘<time.h>’; did you forget to ‘#include <time.h>’?
   12 | #include <unistd.h>
  +++ |+#include <time.h>
   13 | 
src/main.c:21:27: error: storage size of ‘tm’ isn’t known
   21 |                 struct tm tm;
      |                           ^~
src/main.c:22:17: warning: implicit declaration of function ‘localtime_r’ [-Wimplicit-function-declaration]
   22 |                 localtime_r(&now, &tm);
      |                 ^~~~~~~~~~~
src/main.c:32:24: warning: implicit declaration of function ‘asctime’ [-Wimplicit-function-declaration]
   32 |                        asctime(&tm));
      |                        ^~~~~~~
src/main.c:32:24: note: ‘asctime’ is defined in header ‘<time.h>’; did you forget to ‘#include <time.h>’?
make: *** [Makefile.posix:4: gettimeofday] Error 1
```

With a time.h include added, it compiles.
Strangely, when using the makefile, it builds without an error, when building with ninja and cmake though, this error is emitted:
```
/tmp/aa/zephyr/samples/posix/gettimeofday/src/main.c: In function ‘_posix_zephyr_main’:
/tmp/aa/zephyr/samples/posix/gettimeofday/src/main.c:30:71: warning: right shift count >= width of type [-Wshift-count-overflow]
   30 |                        "tv_usec=%d\n\t%s\n", (unsigned int)(tv.tv_sec >> 32),
      |                                                                       ^~
```

**To Reproduce**
Build sample as indicated

**Expected behavior**
Build to work

**Impact**
None

**Environment (please complete the following information):**
 - OS: Arch Linux x86_64
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA or Version used: 24f2c30678eec81095e8f5db977800c317e5ce53

--- a/samples/posix/gettimeofday/src/main.c
+++ b/samples/posix/gettimeofday/src/main.c
@@ -8,6 +8,7 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <sys/time.h>
+#include <time.h>
 #include <unistd.h>
 
 int main(void)

